packages/opentelemetry-sdk-trace-base/src/export/SimpleSpanProcessor.ts

```diff
@@ -57,21 +59,38 @@ export class SimpleSpanProcessor implements SpanProcessor {
       return;
     }
 
-    internal
-      ._export(this._exporter, [span])
-      .then((result: ExportResult) => {
-        if (result.code !== ExportResultCode.SUCCESS) {
-          globalErrorHandler(
-            result.error ??
-              new Error(
-                `SimpleSpanProcessor: span export failed (status ${result})`
-              )
-          );
-        }
-      })
-      .catch(error => {
-        globalErrorHandler(error);
-      });
+    const doExport = () =>
+      internal
+        ._export(this._exporter, [span])
+        .then((result: ExportResult) => {
+          if (result.code !== ExportResultCode.SUCCESS) {
+            globalErrorHandler(
+              result.error ??
+                new Error(
+                  `SimpleSpanProcessor: span export failed (status ${result})`
+                )
+            );
+          }
+        })
+        .catch(error => {
+          globalErrorHandler(error);
+        });
+
+    // Avoid scheduling a promise to make the behavior more predictable and easier to test
+    if (span.resource.asyncAttributesPending) {
+      const exportPromise = span.resource.waitForAsyncAttributes().then(
+        () => {
+          this._unresolvedExports.delete(exportPromise);
+          return doExport();
```

Comment:
I think the promise is removed from `_unresolvedExports` before calling the `doExport`, so the `forceFlush` can return before the export is actually done